44.md

```diff
@@ -0,0 +1,80 @@
+
+**`tags`** MUST contain an entry identifying the receiver of the message (such that relays may naturally forward this event to them), in the form `["p", "<pubkey, as a hex string>"]`.
+
+**`tags`** MAY contain an entry identifying the previous message in a conversation or a message we are explicitly replying to (such that contextual, more organized conversations may happen), in the form `["e", "<event_id>"]`.
+
+**Note**: By default in the [libsecp256k1](https://github.com/bitcoin-core/secp256k1) ECDH implementation, the secret is the SHA256 hash of the shared point (both X and Y coordinates). We are using this exact implementation. In NIP4, unhashed shared point was used.
+
+## Versioning
+
+Clients MUST throw a descriptive error if they receive NIP44 message, version of which they don't support.
+
+Currently defined encryption algorithms:
+
+- `0x00` - RESERVED
+- `0x01` - XChaCha20 + sha256(ecdh)
+
+## Security Warning
+
+This standard does not go anywhere near what is considered the state-of-the-art in encrypted communication between peers, and it leaks metadata in the events, therefore it must not be used for anything you really need to keep secret, and only with relays that use `AUTH` to restrict who can fetch your `kind:4` events.
+
+## Client Implementation Warning
+
+Clients *should not* search and replace public key or note references from the `.content`. If processed like a regular text note (where `@npub...` is replaced with `#[0]` with a `["p", "..."]` tag) the tags are leaked and the mentioned user will receive the message in their inbox.
+
+## Algorithm
+
+```js
+// npm install @noble/curves @noble/hashes @scure/base @stablelib/xchacha20
+import {secp256k1} from '@noble/curves/secp256k1'
+import {sha256} from '@noble/hashes/sha256'
+import {randomBytes} from '@noble/hashes/utils'
+import {base64} from '@scure/base'
+import {streamXOR as xchacha20} from '@stablelib/xchacha20'
+
+export function getConversationKey(privkeyA: string, pubkeyB: string) {
+  const key = secp256k1.getSharedSecret(privkeyA, '02' + pubkeyB)
+  return sha256(key.slice(1, 33))
+}
+
+export function encrypt(
+  privkey: string,
+  pubkey: string,
+  text: string,
+  ver = 1
+): string {
+  if (ver !== 1) throw new Error('NIP44: unknown encryption version')
+  let key = getConversationKey(privkey, pubkey)
+  let nonce = randomBytes(24)
+  let plaintext = new TextEncoder().encode(text)
+  let ciphertext = xchacha20(key, nonce, plaintext, plaintext)
+  let ctb64 = base64.encode(ciphertext)
+  let nonceb64 = base64.encode(nonce)
+  return JSON.stringify({ciphertext: ctb64, nonce: nonceb64, v: 1})
+}
+
+export function decrypt(privkey: string, pubkey: string, data: string): string {
+  let dt = JSON.parse(data)
+  if (dt.v !== 1) throw new Error('NIP44: unknown encryption version')
```

Comment:
should probably have something in the nip specification that says how user interfaces should deal with versions they don't know about:   let users know they got a message, tell them the version isn't supported and suggest appropriate action (upgrade, etc.)

Reply:
Agree.